selfdrive/car/gm/values.py

```diff
@@ -49,6 +50,16 @@ class CarControllerParams:
   GAS_LOOKUP_V = [MAX_ACC_REGEN, ZERO_GAS, MAX_GAS]
   BRAKE_LOOKUP_V = [MAX_BRAKE, 0.]
 
+  # determined by letting Volt regen to a stop in L gear from 89mph,
+  # and by letting off gas and allowing car to creep, for determining
+  # the positive threshold values at very low speed
+  EV_GAS_BRAKE_THRESHOLD_BP = [0.5, 0.6, 0.7, 0.8, 0.9, 1.0, 1.1, 1.2, 1.29, 1.52, 1.55, 1.6, 1.7, 1.8, 2.0, 2.2, 2.5, 5.52, 9.6, 20.5, 23.5, 35.0] # [m/s]
+  EV_GAS_BRAKE_THRESHOLD_V = [0.678, 0.67, 0.63, 0.56, 0.49, 0.38, 0.3, 0.13, 0.0, -0.14, -0.16, -0.18, -0.215, -0.255, -0.32, -0.41, -0.5, -0.72, -0.895, -1.125, -1.145, -1.16] # [m/s^s]
```

Comment:
Any chance we can make this a simper/linear piecewise function?

Reply:
Is L gear just requesting full regen with no friction brakes?

Reply:
@twilsonco can you gather this data again, but instead just request max regen with no brake? Then we can guarantee these values are correct (rather than the L gear giving us a different response as @hewers points out)

Reply:
This also doesn't match what you wrote above where you claim 2.89 m/s is where regen accel goes to 0 m/s/s

Reply:
> This also doesn't match

Correct. As I continue to drive on this on my fork, I found that the positive acceleration portion of the gas/brake profile made the last bit of braking to a stop harsher than preferred. So the current profile in my fork modifies this in two ways; 1) removing the higher positive acceleration values at low speed, and moving the 0m/s^2 threshold to a lower speed. 

I made these changes _without followup tests to generation new profiles_. Rather I made the change based on observation and logs and verified they improved stopping performance anecdotally. https://github.com/twilsonco/openpilot/blob/18ddb5f231b88b5bb86042394ba273adfb1d4000/selfdrive/car/gm/values.py#L71

Reply:
> @twilsonco can you gather this data again, but instead just request max regen with no brake?

I don't follow. These were recorded in L mode with no brakes. This is the amount of regen accessible to openpilot via gas command when in D or L mode, and the only way to get more regen is with the regen-on-demand paddle which we can't operate from openpilot. Otherwise, the braking system _will_ blend in as much regen as it can when you hit the brakes, but that's also the case with brake commands from openpilot, and it has nothing to do with the amount of regen accessible when the brakes aren't being used.

Now, we did find that a brake command of, say, 300, produces more brake force at 40mph than at 10mph, which seems to be a consequence of the lack of available regen at lower speeds, and can be fixed by increasing the max brake value at lower speeds, then MAX\_BRAKE will always produce the expected -3.5m/s^2 of brake. I do this in my fork already https://github.com/twilsonco/openpilot/blob/18ddb5f231b88b5bb86042394ba273adfb1d4000/selfdrive/car/gm/values.py#L77

Reply:
I'm just saying you should gather the data again using the regen command, vs the L gear so we know for sure the observations are correct. Did you just find that L gear is similar to the regen command and use that?

Reply:
Feel free to update the simpler with your updated observations btw!

Reply:
> I'm just saying you should gather the data again using the regen command, vs the L gear so we know for sure the observations are correct. Did you just find that L gear is similar to the regen command and use that?

Yes, I used L mode under the observation-based assumption that it has access to the same amount of regen as openpilot when using the MAX\_REGEN value. It may be worth it to rerun the tests using a hardcoded MAX\_REGEN command just to be sure, as you suggest.